src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -5225,7 +5207,8 @@ IArrayInitializerOperation popAndAssembleArrayInitializerValues(IArrayInitialize
 
         public override IOperation VisitInstanceReference(IInstanceReferenceOperation operation, int? captureIdForResult)
         {
-            if (operation.ReferenceKind == InstanceReferenceKind.ImplicitReceiver)
+            if (operation.ReferenceKind == InstanceReferenceKind.ImplicitReceiver &&
+                !operation.Type.IsAnonymousType)
```

Comment:
>!operation.Type.IsAnonymousType [](start = 16, length = 31)

I am not sure why this condition is added. Also, I believe the behavior will be incorrect for the case when an instance of anonymous type is used as a target of a With statement in VB. Please add a test.

Reply:
Added test `WithFlow_07` which verifies the old baseline was incorrect and removing this check produces the correct baseline.

---
In reply to: [197208455](https://github.com/dotnet/roslyn/pull/28005#discussion_r197208455) [](ancestors = 197208455)